File: Plugins/VanillaBasics/src/main/java/org/tobi29/scapes/vanilla/basics/util/ToolUtil.java
/*
 * Copyright 2012-2015 Tobi29
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.tobi29.scapes.vanilla.basics.util;

import org.tobi29.scapes.block.ItemStack;
import org.tobi29.scapes.vanilla.basics.VanillaBasics;
import org.tobi29.scapes.vanilla.basics.material.AlloyType;
import org.tobi29.scapes.vanilla.basics.material.VanillaMaterial;
import org.tobi29.scapes.vanilla.basics.material.item.ItemIngot;

public class ToolUtil {
    public static boolean createTool(VanillaBasics plugin, ItemStack item,
            String type) {
        return createTool(plugin, item, id(type));
    }

    public static boolean createTool(VanillaBasics plugin, ItemStack item,
            int id) {
        VanillaMaterial materials = plugin.getMaterials();
        ItemIngot ingot = (ItemIngot) item.material();
        MetalUtil.Alloy alloy = ingot.alloy(item);
        AlloyType alloyType = alloy.type(plugin);
        double efficiency = alloyType.baseToolEfficiency();
        double strength = alloyType.baseToolStrength();
        double damage = alloyType.baseToolDamage();
        int level = alloyType.baseToolLevel();
        switch (id) {
            case 1:
                item.setMaterial(materials.metalPickaxe);
                strength *= 0.1;
                break;
            case 2:
                item.setMaterial(materials.metalAxe);
                strength *= 1.5;
                break;
            case 3:
                item.setMaterial(materials.metalShovel);
                strength *= 0.1;
                break;
            case 4:
                item.setMaterial(materials.metalHammer);
                efficiency = 0.0;
                strength *= 0.4;
                break;
            case 5:
                item.setMaterial(materials.metalSaw);
                strength *= 0.1;
                break;
            case 6:
                item.setMaterial(materials.metalHoe);
                strength *= 0.1;
                break;
            case 7:
                item.setMaterial(materials.metalSword);
                efficiency = 0.0;
                break;
            default:
                return false;
        }
        item.metaData("Vanilla").setDouble("ToolEfficiency", efficiency);
        item.metaData("Vanilla").setDouble("ToolStrength", strength);
        item.metaData("Vanilla").setDouble("ToolDamageAdd", damage);
        item.metaData("Vanilla").setInteger("ToolLevel", level);
        return true;
    }

    public static boolean createStoneTool(VanillaBasics plugin, ItemStack item,
            String type) {
        return createStoneTool(plugin, item, id(type));
    }

    public static boolean createStoneTool(VanillaBasics plugin, ItemStack item,
            int id) {
        VanillaMaterial materials = plugin.getMaterials();
        double efficiency = 1.0;
        double strength = 4.0;
        double damage = 0.004;
        int level = 10;
        switch (id) {
            case 1:
                item.setMaterial(materials.stonePickaxe);
                strength *= 0.1;
                break;
            case 2:
                item.setMaterial(materials.stoneAxe);
                strength *= 1.5;
                break;
            case 3:
                item.setMaterial(materials.stoneShovel);
                strength *= 0.1;
                break;
            case 4:
                item.setMaterial(materials.stoneHammer);
                efficiency = 0.0;
                strength *= 0.4;
                break;
            case 5:
                item.setMaterial(materials.stoneSaw);
                strength *= 0.1;
                break;
            case 6:
                item.setMaterial(materials.stoneHoe);
                strength *= 0.1;
                break;
            case 7:
                item.setMaterial(materials.stoneSword);
                efficiency = 0.0;
                break;
            default:
                return false;
        }
        item.metaData("Vanilla").setDouble("ToolEfficiency", efficiency);
        item.metaData("Vanilla").setDouble("ToolStrength", strength);
        item.metaData("Vanilla").setDouble("ToolDamageAdd", damage);
        item.metaData("Vanilla").setInteger("ToolLevel", level);
        return true;
    }

    public static int id(String type) {
        switch (type) {
            case "Pickaxe":
                return 1;
            case "Axe":
                return 2;
            case "Shovel":
                return 3;
            case "Hammer":
                return 4;
            case "Saw":
                return 5;
            case "Hoe":
                return 6;
            case "Sword":
                return 7;
            default:
                return -1;
        }
    }
}

File: Plugins/VanillaBasics/src/main/kotlin/scapes/plugin/tobi29/vanilla/basics/util/ToolUtil.kt
/*
 * Copyright 2012-2017 Tobi29
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package scapes.plugin.tobi29.vanilla.basics.util

import org.tobi29.scapes.block.ItemStack
import org.tobi29.scapes.engine.utils.io.tag.set
import scapes.plugin.tobi29.vanilla.basics.VanillaBasics
import scapes.plugin.tobi29.vanilla.basics.material.item.ItemIngot

fun createTool(plugin: VanillaBasics,
               item: ItemStack,
               type: String): Boolean {
    return createTool(plugin, item, id(type))
}

fun createTool(plugin: VanillaBasics,
               item: ItemStack,
               id: Int): Boolean {
    val materials = plugin.materials
    val ingot = item.material() as ItemIngot
    val alloy = ingot.alloy(item)
    val alloyType = alloy.type(plugin)
    var efficiency = alloyType.baseToolEfficiency()
    var strength = alloyType.baseToolStrength()
    val damage = alloyType.baseToolDamage()
    val level = alloyType.baseToolLevel()
    when (id) {
        1 -> {
            item.setMaterial(materials.metalPickaxe)
            strength *= 0.1
        }
        2 -> {
            item.setMaterial(materials.metalAxe)
            strength *= 1.5
        }
        3 -> {
            item.setMaterial(materials.metalShovel)
            strength *= 0.1
        }
        4 -> {
            item.setMaterial(materials.metalHammer)
            efficiency = 0.0
            strength *= 0.4
        }
        5 -> {
            item.setMaterial(materials.metalSaw)
            strength *= 0.1
        }
        6 -> {
            item.setMaterial(materials.metalHoe)
            strength *= 0.1
        }
        7 -> {
            item.setMaterial(materials.metalSword)
            efficiency = 0.0
        }
        else -> return false
    }
    item.metaData("Vanilla")["ToolEfficiency"] = efficiency
    item.metaData("Vanilla")["ToolStrength"] = strength
    item.metaData("Vanilla")["ToolDamageAdd"] = damage
    item.metaData("Vanilla")["ToolLevel"] = level
    return true
}

fun createStoneTool(plugin: VanillaBasics,
                    item: ItemStack,
                    type: String): Boolean {
    return createStoneTool(plugin, item, id(type))
}

fun createStoneTool(plugin: VanillaBasics,
                    item: ItemStack,
                    id: Int): Boolean {
    val materials = plugin.materials
    var efficiency = 1.0
    var strength = 4.0
    val damage = 0.004
    val level = 10
    when (id) {
        1 -> {
            item.setMaterial(materials.flintPickaxe)
            strength *= 0.1
        }
        2 -> {
            item.setMaterial(materials.flintAxe)
            strength *= 1.5
        }
        3 -> {
            item.setMaterial(materials.flintShovel)
            strength *= 0.1
        }
        4 -> {
            item.setMaterial(materials.flintHammer)
            efficiency = 0.0
            strength *= 0.4
        }
        5 -> {
            item.setMaterial(materials.flintSaw)
            strength *= 0.1
        }
        6 -> {
            item.setMaterial(materials.flintHoe)
            strength *= 0.1
        }
        7 -> {
            item.setMaterial(materials.flintSword)
            efficiency = 0.0
        }
        else -> return false
    }
    item.metaData("Vanilla")["ToolEfficiency"] = efficiency
    item.metaData("Vanilla")["ToolStrength"] = strength
    item.metaData("Vanilla")["ToolDamageAdd"] = damage
    item.metaData("Vanilla")["ToolLevel"] = level
    return true
}

fun id(type: String): Int {
    when (type) {
        "Pickaxe" -> return 1
        "Axe" -> return 2
        "Shovel" -> return 3
        "Hammer" -> return 4
        "Saw" -> return 5
        "Hoe" -> return 6
        "Sword" -> return 7
        else -> return -1
    }
}
